src/temple/front/block-explorers.ts

```diff
@@ -146,41 +181,42 @@ export function useChainBlockExplorers(chainKind: TempleChainKind, chainId: stri
       genericRemoveBlockExplorers(
         chainKind,
         chainId,
-        chainBlockExplorers.map(({ id }) => id)
+        getBlockExplorers(String(chainId)).map(({ id }) => id)
       ),
-    [chainBlockExplorers, chainId, chainKind, genericRemoveBlockExplorers]
-  );
-
-  const activeBlockExplorer = useMemo<BlockExplorer | undefined>(
-    () => chainBlockExplorers.find(({ id }) => id === activeBlockExplorerId) ?? chainBlockExplorers[0],
-    [activeBlockExplorerId, chainBlockExplorers]
+    [getBlockExplorers, chainId, chainKind, genericRemoveBlockExplorers]
   );
 
   return {
-    chainBlockExplorers,
-    activeBlockExplorer,
     addBlockExplorer,
     removeBlockExplorer,
     removeAllBlockExplorers,
     replaceBlockExplorer
   };
 }
 
+/** (!) Very expensive hook for lists */
```

Comment:
I believe that it's true but could you please explain in this comment what's wrong with it's implementation in this thread, if you know?